cd/python/docker/python_images.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Executes mxnet python images pipeline functions: build, test, publish
+# Assumes script is run from the root of the mxnet repository
+# Assumes script is being run within MXNet CD infrastructure
+
+set -xe
+
+usage="Usage: python_images.sh <build|test|publish> MXNET-VARIANT <py2|py3>"
+
+command=${1:?$usage}
+mxnet_variant=${2:?$usage}
+python_version=${3:?usage}
+
+cd_utils='cd/utils'
+
+case ${python_version} in
+    py3)
+        python_cmd="python3"
+        ;;
+    py2)
+        python_cmd="python"
+        ;;
+    *)
+        echo "Error: specify python version with either 'py2' or 'py3'"
+        exit 1
+        ;;
+esac
+
+docker_tags=($(./${cd_utils}/docker_tag.sh ${mxnet_variant}))
+main_tag="${docker_tags[0]}_${python_version}"
+base_image=$(./${cd_utils}/mxnet_base_image.sh ${mxnet_variant})
+repository="python"
+image_name="${repository}:${main_tag}"
+
+resources_path='cd/python/docker'
+
+if [ ! -z "${RELEASE_DOCKERHUB_REPOSITORY}" ]; then
+    image_name="${RELEASE_DOCKERHUB_REPOSITORY}/${image_name}"
+fi
+
+# Builds runtime image
+build() {
+    docker build -t "${image_name}" --build-arg PYTHON_CMD=${python_cmd} --build-arg BASE_IMAGE="${base_image}" --build-arg MXNET_COMMIT_ID=${GIT_COMMIT} -f ${resources_path}/Dockerfile .
```

Comment:
Raw docker access is highly discouraged. Please use our Python Docker wrapper. Otherwise you are doing to run into all sorts of leaks

Reply:
What will leak here? This assumes docker login has already been called

Reply:
Docker build and docker run tend to leak resources and thus leave behind unclaimed containers. It took us a lot of time to get it right

Reply:
What will leak here? This assumes docker login has already been called

Reply:
I get it - but it's way to tightly integrated with the CI stuff. I don't think it should be done here.

Reply:
As a compromise, I've added a trap to the script. This does basically what the build.py does.

Reply:
We spent months on fixing all the problems that arose, so please spend a bit of time on integrating your stuff. We don't want the same issues to arise again - the current solution is proven to work.

Reply:
Unfortunately, I don't have the time to refactor build.py and extract that functionality and make it general purpose. If you are willing, I'm happy to refactor this stuff once it is done...

Reply:
The time saved now is paid by increased maintenance overhead and risk due to two separate solutions. So this is a time trade, where the increased maintenance overhead will then have to be handled by the community - sorry, but I don't like that stance.